File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/system-admin/shared/database-log.model.ts
import {
    NgForm,
    FormGroup,
    FormControl,
    Validators,
    FormBuilder
} from '@angular/forms'
import * as moment from 'moment/moment';

export class DatabaseLogModel {
        public  DBLogId :number=0;
        public FileName: string = null;
        public FolderPath: string = null;
        public DatabaseName: string = null;
        public DatabaseVersion: string = null;
        public IsDBRestorable: boolean = false;
        public Action: string = null;
        public ActionType: string = null;
        public Status: string = null;
        public MessageDetail: string = null;
        public Remarks: string = null;
        public CreatedBy: number = null;
        public CreatedOn: string = null;
        public IsActive: boolean = null;

        public DBLogValidator: FormGroup = null;

        public IsDirty(fieldname): boolean {
            if (fieldname == undefined) {
                return this.DBLogValidator.dirty;
            }
            else {
                return this.DBLogValidator.controls[fieldname].dirty;
            }

        }

        public IsValid():boolean{if(this.DBLogValidator.valid){return true;}else{return false;}} 
        public IsValidCheck(fieldname, validator): boolean {            
            if (this.DBLogValidator.valid) {
                return true;
            }
          
            if (fieldname == undefined) {
                return this.DBLogValidator.valid;
            }
            else {

                return !(this.DBLogValidator.hasError(validator, fieldname));
            }
        }

        constructor() {
            var _formBuilder = new FormBuilder();
            this.DBLogValidator = _formBuilder.group({
                'Remarks': ['', Validators.compose([Validators.required, Validators.maxLength(300)])],              
            });
        }
           
}

File: Code/Components/DanpheEMR.ServerModel/SystemAdminModels/DatabaseLogModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DanpheEMR.ServerModel
{
    public class DatabaseLogModel
    {
        [Key]
        public int DBLogId { get; set; }
        public string FileName { get; set; }
        public string FolderPath { get; set; }
        public string DatabaseName { get; set; }
        public string DatabaseVersion { get; set; }
        public Boolean? IsDBRestorable { get; set; }
        public string Action { get; set; }
        public string ActionType { get; set; }
        public string Status  { get; set; }
        public string MessageDetail { get; set; }
        public string Remarks { get; set; }
        public int?  CreatedBy { get; set; }
        public DateTime? CreatedOn { get; set; }
        public DateTime? DeleteOn { get; set; }
        public Boolean? IsActive { get; set; }     		
    }
}
